sklearn/preprocessing/data.py

```diff
@@ -323,7 +323,7 @@ def transform(self, X):
         """
         check_is_fitted(self, 'scale_')
 
-        X = check_array(X, copy=self.copy, ensure_2d=False)
+        X = check_array(X, copy=self.copy, ensure_2d=False, dtype=FLOAT_DTYPES)
```

Comment:
> You get an array of twice the memory in an 'int32' cast to 'float64' case but
> this may not happen that much in practice I guess.

In principle, I agree with you. But here we are trying to get a hot fix
and not to delay the release. Addressing your comment would require
developping new validation code.